Maman11/Q1/my_add.c

```diff
@@ -1,25 +1,10 @@
 #include <stdio.h>
 #include <stdlib.h>
+#define DATA_SIZE 33
 
-int main() {
-    int num1, num2, binary1, binary2;
-    printf("Enter a number: ");
-    scanf("%d", &num1); // Reads an integer from the user
-
-    printf("Enter a second number: ");
-    scanf("%d", &num2); // Reads an integer from the user
-
-    printf("\nYou entered : %d and %d\n\n", num1, num2);
-
-    binary1 = decimalToBinary(num1);
-    binary2 = decimalToBinary(num2);
-    printf("\nThe sum of %d and %d is: %d or %d in decimal\n", num1, num2, addBinary(binary1, binary2), num1+num2);
-    return 0;
-}
-
-int decimalToBinary(int num){
+unsigned int decimalToBinary(int num){
     int result = num;
-    char binary[33];
+    char binary[DATA_SIZE];
     int index = 0;
 
     while (result > 0) {
```

Comment:
I cannot write a review on the code block below but
on line wiht '\0' you should use a defin for it.

Using. define makes the code clearer.
Think about this:
#define EOS '\0' // (EOS = End Of String)

Then in every line you need \0 it will be clearer what is \0